packages/yarnpkg-core/sources/scriptUtils.ts

```diff
@@ -377,7 +377,7 @@ export async function prepareExternalProject(cwd: PortablePath, outputPath: Port
             // We can't use `npm ci` because some projects don't have npm
             // lockfiles that are up-to-date. Hopefully npm won't decide
             // to change the versions randomly.
-            const install = await execUtils.pipevp(`npm`, [`install`], {cwd, env, stdin, stdout, stderr, end: execUtils.EndStrategy.ErrorCode});
+            const install = await execUtils.pipevp(`npm`, [`install`, `--legacy-peer-deps`], {cwd, env, stdin, stdout, stderr, end: execUtils.EndStrategy.ErrorCode});
```

Comment:
I added this flag because I noticed at work that we depended on git packages that couldn't be installed anymore due to `ERESOLVE` errors.

Whether we should keep it or not is debatable ... new npm packages that rely on the broken auto-install behaviour will cease to work, but older npm packages that relied on the correct behaviour will work again.

I'd tend to go with the later for two reasons: what used to work should keep working, and since it matches the correct behaviour (and ours), it's probably the one we should "reward". People relying on peer dependencies' auto-install are knowingly writing non-portable packages, they should know better.

Perhaps, optionally, we could also check whether there's a `packageManager` field set to npm >=7, and remove the flag in this case, to try to have a "best of both worlds".